YellowNotes.Api/Controllers/NoteController.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Services;
+using System.Threading;
+using YellowNotes.Core.Utility;
+using System.Collections.Generic;
+
+namespace YellowNotes.Api.Controllers
+{
+    [ApiController]
+    [Route("notes")]
+    public class NoteController : ControllerBase
+    {
+        private readonly INoteService noteService;
+
+        public NoteController(INoteService noteService) => this.noteService = noteService;
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<NoteDto>>> GetNotes([FromQuery] int takeCount = 20, [FromQuery] int skipCount = 0,
+            CancellationToken cancellationToken = default)
+        {
+            var userEmail = HttpContext.GetEmailFromClaims();
+            if (userEmail == null)
+            {
+                return BadRequest("Error: HttpContext.User.EmailClaim is null");
+            }
+
+            var errorMessage = TokenUtility.Authorize(userEmail, Request.Headers);
+            if (errorMessage != null)
+            {
+                return Unauthorized(errorMessage);
+            }
+            var notes = await noteService.GetNotes(takeCount, skipCount, userEmail, cancellationToken);
+            return Ok(notes);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateNote([FromBody] NoteDto noteDto,
+            CancellationToken cancellationToken = default)
+        {
+            var userEmail = HttpContext.GetEmailFromClaims();
+            if (userEmail == null)
+            {
+                return BadRequest("Error: HttpContext.User.EmailClaim is null");
+            }
+
+            var errorMessage = TokenUtility.Authorize(userEmail, Request.Headers);
+            if (errorMessage != null)
+            {
+                return Unauthorized(errorMessage);
+            }
+
+            var success = await noteService.CreateNote(noteDto, userEmail, cancellationToken);
+            if (!success)
+            {
+                return BadRequest("Failed to create note");
+            }
+
+            return NoContent();
```

Comment:
after creation of a new resource, typically api returns url to that resource.
use `CreatedAtAction()`